src/rpc/mining.cpp

```diff
@@ -787,8 +787,14 @@ UniValue submitblock(const JSONRPCRequest& request)
     if (!DecodeHexBlk(block, request.params[0].get_str()))
         throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "Block decode failed");
 
-    if (block.vtx.empty() || !block.vtx[0]->IsCoinBase()) {
-        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "Block does not start with a coinbase");
+    if (block.vtx.empty())
+    {
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "Block does not contain a coinbase transaction");
```

Comment:
I would change a bit the phrasing to emphasis that there is no transactions at all and that a valid block **has** to have at least one.

Something like this ?
"Block does not contain any transaction. Coinbase transaction required."

Reply:
> [..] and that a valid block has to have at least one.

A valid block has to have *at least a coinbase transaction*, not just any transaction. I think your suggested improvement is fine though because it gives the solution.

Reply:
I tried a few phrasings, but don't like any of these:

* Block does not contain any transactions - isn't specific enough about what to fix
* Block is empty - definitely not specific enough about why this is a problem
* Block must contain a coinbase tranasction - potentially better but I'm really nit picking

I think this is fine, basically, and we can tweak later.